sycl/include/sycl/ext/oneapi/kernel_properties/properties.hpp

```diff
@@ -60,13 +60,15 @@ struct device_has_key
                                  std::integral_constant<aspect, Aspects>...>;
 };
 
-struct nd_range_kernel_key {
+struct nd_range_kernel_key
+    : detail::compile_time_property_key<detail::PropKind::NDRangeKernel> {
```

Comment:
this is a bug-fix really, the issue has been uncovered by one of the `static_assert`s that I'm adding here.